functional-test/src/test/groovy/org/rundeck/util/gui/pages/activity/ActivityPage.groovy

```diff
@@ -44,4 +45,11 @@ class ActivityPage extends BasePage implements ActivityListTrait{
         els By.xpath("//*[contains(text(),'${jobName}')]")
     }
 
+    static void validateFirstActivityRow(List<WebElement> activityList) {
+        assert !activityList.isEmpty() : "Activity list is empty, no rows to validate."
```

Comment:
Assertions are for tests, but this is the "model".
You're using groovy assertions here (that are different from Spec assertions) that are generally used to asserts invariants -- e.g. null checks.